pkg/controller/job/job_controller.go

```diff
@@ -1761,3 +1773,16 @@ func countReadyPods(pods []*v1.Pod) int32 {
 	}
 	return cnt
 }
+
+// This checks if we should apply PodRecreationPolicy.
+// PodRecreationPolicy controls when we recreate pods if they are marked as terminating
+// Failed means that we recreate only once the pod has terminated.
+func hasPodReplacementPolicyFailed(job *batch.Job) bool {
+	return feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) && *job.Spec.PodReplacementPolicy == batch.Failed
+}
+
+// In PodRecreationPolicy we enable counting of terminating pods.
+// We will enable this in case of JobPodReplacementPolicy being enabled
+func shouldCountTerminating(job *batch.Job) bool {
```

Comment:
is it necessary to have a wrapper function for this one? Especially since it is called only once?

Reply:
I updated this and we are now using this function twice.

Reply:
even if it's called twice, it's so simple that I would prefer calling `Enabled` directly, to make it explicit that this is about gating.